### DISCUSSION TITLE
Insert link to notebook without embedding any cell

### DISCUSSION THREAD

**Discussion by Elle518** [quarto-dev/quarto-cli#6180] [Q&A]
### Description

Hello!

I want to insert a link to a notebook inside a book chapter but without embedding any particular cell of that notebook. Is that possible?

I've been reading the documentation regarding to "Embedding Jupyter Notebook Cells" and "Links to Source Notebooks", but I haven't been able to just link a notebook. My notebook is in the book project inside a folder called "notebooks".

Thank you in advance!

**Comment by cscheid** [comment]
You should be able to use a regular markdown link, like so: `[my notebook](/notebooks/some-notebook.ipynb)`.

I don't think we do automatic resource discovery for `.ipynb` files, so you'll also need to add the notebook folder (or just the specific notebook) to the [`resources` field](https://quarto.org/docs/websites/website-tools.html#site-resources).

(This link, of course, won't work in the PDF format of the book, because what would it link to?)

**Reply by Elle518** [reply]
Hello!

Thank you very much for responding so quickly. I have tried the solution that you propose, and it effectively links to the notebook. The problem is that it does not show a version rendered in HTML like the one shown in the documentation in the "Notebooks Views" section, but instead shows the JSON behind the notebook.

I've included this to the _quarto.yml of the project:
```
project:
  type: book
  resources:
    - "*.ipynb"
```

Is there a way to make that link lead to a rendered HTML version of the notebook as in "Notebook Views"?

Thanks!

**Reply by cscheid** [reply]
> Is there a way to make that link lead to a rendered HTML version of the notebook as in "Notebook Views"?

You will need to render the notebook to HTML. `quarto render` takes `.ipynb` files as well. See these:

- https://quarto.org/docs/projects/code-execution.html#notebooks
- https://quarto.org/docs/tools/vscode-notebook.html#render-and-preview (this link is specific to vscode but the feature is general.)

**Reply by dragonstyle** [reply]
I've been thinking about whether it makes sense to support embedding only a link to a cell (not the rendered output). One alternative that you can consider is to use a `manuscript` project if it is suitable.  That type of project (new in Quarto 1.4) does quite a bit more orchestration of notebooks with a root article and includes automatically resolving notebook links like @cscheid suggested into their previews (and generating the previews and so on).

More about this project type here:
https://quarto.org/docs/manuscripts/

If that doesn't look suitable for what you're doing, could you tel me a bit more about the scenario - it will be useful to have real world examples as I think through any linking type embed behavior!

(edit)

To be more explicit - in a manuscript project the link:

```markdown
[my notebook](/notebooks/some-notebook.ipynb)
```

Would automatically link to a generated HTML preview.

**Reply by Elle518** [reply]
Thank you both very much for your answers! Sorry for the delay in responding. I have tried to render the html as suggested by cscheid and it renders perfectly. The problem arises when I try to access the redendered notebook from the link created inside the .qmd file like this:

`[MyNotebook](/notebooks/my_notebook.html)`

When I click on that link, the html opens but it is not displayed correctly in the browser and in the console I get error messages of the type:

```
GET: /notebooks/my_notebook.html
  /notebooks/my_notebook_files/libs/clipboard/clipboard.min.js (404: Not found)
  /notebooks/my_notebook_files/libs/quarto-html/quarto.js (404: Not found)
  ...
```
But these files are indeed in the project.

What would then be the correct way to open that rendered notebook from a .qmd?

Thanks!
